include/libdnf/base/transaction.hpp

```diff
@@ -64,6 +74,33 @@ class Transaction {
     // @replaces libdnf/Goal.formatAllProblemRules(const std::vector<std::vector<std::string>> & problems);
     std::string all_package_solver_problems_to_string();
 
+    /// Retrive comment describing the transaction
+    std::optional<std::string> get_comment() const noexcept;
+
+    /// Set a user-specified comment describing the transaction
+    void set_comment(const std::string & value);
+
+    /// Get command line of a program that started the transaction
+    std::optional<std::string> get_cmdline() const noexcept;
+
+    /// Set command line of a program that started the transaction
+    void set_cmdline(const std::string & value);
```

Comment:
In case I am using GUI (Gnome Software), what will be expected to be stored here?

Reply:
I guess that we use `cmdline` as a label of particular transaction. Mostly it contains cmdline command.